pkg/resource/nodecadaemon.go

```diff
@@ -49,6 +49,33 @@ spec:
         - "/bin/sh"
         - "-c"
         - |
+          for f in $(ls /tmp/serviceca); do
+              if [ "${f}" == "service-ca.crt" ]; then
+                  continue
+              fi
+              echo $f
+              ca_file_path="/tmp/serviceca/${f}"
+              reg_dir_path="/etc/docker/certs.d/${f}"
+              if [ -e "${reg_dir_path}" ]; then
+                  cp -u $ca_file_path $reg_dir_path/ca.crt
+              else
+                  mkdir $reg_dir_path
+                  cp $ca_file_path $reg_dir_path/ca.crt
+              fi
+          done
+          for d in $(ls /etc/docker/certs.d); do
+              echo $d
+              if [ "${d}" == "image-registry.openshift-image-registry.svc:5000" ]; then
+                  continue
+              fi
+              if [ "${d}" == "image-registry.openshift-image-registry.svc.cluster.local:5000" ]; then
+                  continue
+              fi
+              reg_conf_path="/tmp/serviceca/${d}"
+              if [ ! -e "${reg_conf_path}" ]; then
+                  rm -rf /etc/docker/certs.d/$d
+              fi
+          done
```

Comment:
you can't do this just once, you need to keep looping and re-checking periodically.  The pod is not going to get restarted just because the configmap content is updated.  That's why the logic below is a continuous while loop with a sleep.

Reply:
Yeah, I intended to put it in the loop of course :) Fixing.